src/session_impl.cpp

```diff
@@ -6527,6 +6531,17 @@ namespace {
 	}
 #endif
 
+	void session_impl::start_ip_notifier()
+	{
+		INVARIANT_CHECK;
+
+		if (m_ip_notifier) return;
+
+		m_ip_notifier = create_ip_notifier(m_io_service);
+		m_ip_notifier->async_wait([this](error_code const& e)
```

Comment:
is ``m_ip_notifier`` a ``shared_ptr``? if so, perhaps a copy of the shared_ptr should be captured here, just to not have dangling pointers appear at all.